hs_file_types/models/raster.py

```diff
@@ -248,146 +260,241 @@ def supports_delete_folder_on_zip(self):
-    """ raster_file is the temp file retrieved from irods and stored on temp dir in django """
+def raster_file_validation(raster_file, raster_folder=None, resource=None):
+    """ Validates if the relevant files are valid for raster aggregation or raster resource type
+
+    :param  raster_file: a temp file (extension tif or zip) retrieved from irods and stored on temp
+    dir in django
+    :param  raster_folder: (optional) folder in which raster file exists on irods.
+    :param  resource: (optional) an instance of CompositeResource in which raster_file exits.
+    If a value for raster_folder is specified then a value for resource must be specified.
+    :return A list of error messages and a list of file paths for all files that belong to raster
+    """
 
     error_info = []
     new_resource_files_to_add = []
 
     file_name_part, ext = os.path.splitext(os.path.basename(raster_file))
+    ext = ext.lower()
+    create_vrt = True
     if ext == '.tif':
-        # create the .vrt file
-        try:
-            temp_vrt_file_path = create_vrt_file(raster_file)
-        except Exception as ex:
-            error_info.append(ex.message)
-        else:
-            if os.path.isfile(temp_vrt_file_path):
-                new_resource_files_to_add.append(temp_vrt_file_path)
-                new_resource_files_to_add.append(raster_file)
+        if raster_folder is not None:
+            res_files = ResourceFile.list_folder(resource=resource, folder=raster_folder,
+                                                 sub_folders=False)
+
+            # check if there is already a vrt file in that folder
+            vrt_files = [f for f in res_files if f.extension.lower() == ".vrt"]
+            tif_files = [f for f in res_files if f.extension.lower() == ".tif"]
+            if vrt_files:
+                if len(vrt_files) > 1:
+                    error_info.append("More than one vrt file was found.")
+                    return error_info, new_resource_files_to_add
+                create_vrt = False
+            elif len(tif_files) != 1:
+                # if there are more than one tif file, there needs to be one vrt file
+                error_info.append("A vrt file is missing.")
+                return error_info, new_resource_files_to_add
+
+            # get all the resource files from irods to the temp dir where
+            # the temp raster file already exist
+            temp_dir = os.path.dirname(raster_file)
+            for f in res_files:
+                if not raster_file.endswith(f.file_name):
+                    temp_file = utils.get_file_from_irods(f, temp_dir)
+                    new_resource_files_to_add.append(temp_file)
+
+        if create_vrt:
+            # create the .vrt file
+            try:
+                temp_vrt_file_path = create_vrt_file(raster_file)
+            except Exception as ex:
+                error_info.append(ex.message)
+            else:
+                if os.path.isfile(temp_vrt_file_path):
```

Comment:
Shouldn't we check if there is a vrt file first, before creating one?  It'd avoid the exception.  The way it's currently formatted, the error message being logged isn't necessarily relevant.

Reply:
We are checking if vrt file exists. The variable ```create_vrt``` is set to True only if the vrt file doesn't already exist. The function ```create_vrt_file()``` can raise Exception if gdal fails to create this file which I guess can be due many reasons.